### PR TITLE
 Migrate pallet-democracy benchmarks to benchmark v2 syntax

### PR DESCRIPTION
**Pull request by ndkazu**
# Description

Migrates pallet-democracy benchmarks to benchmark v2 syntax
This is Part of https://github.com/paritytech/polkadot-sdk/issues/6202

### CODE REVIEW

File: substrate/frame/democracy/src/benchmarking.rs [hunk 1]
```diff
@@ -193,8 +215,9 @@ benchmarks! {
 			_ => return Err("referendum not ongoing".into()),
 		};
 		assert_eq!(tally.nays, 1000u32.into(), "changed vote was not recorded");
+		Ok(())
 	}
-
+/*
```
Comment (shawntabrizi) [inline substrate/frame/democracy/src/benchmarking.rs:220]: can you mark the PR as draft until you are ready for review

**Review by shawntabrizi — APPROVED** [review]
only a quick look, but seems reasonable to me

**Comment by ndkazu** [comment]
Label needed for the PR :pray:

**Comment by muharem** [comment]
bot help

**Comment by muharem** [comment]
bot bench substrate-pallet --pallet=pallet-democracy

**Comment by muharem** [comment]
bot bench substrate-pallet --pallet=pallet_democracy

**Comment by ndkazu** [comment]
One more approval, and I think we're ready to merge 🤔

**Comment by ndkazu** [comment]
> @muharem Command `"$PIPELINE_SCRIPTS_DIR/commands/bench/bench.sh" --subcommand=pallet --runtime=dev --target_dir=substrate --features=riscv --pallet=pallet_democracy` has finished. Result: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/7782688 has finished. If any artifacts were generated, you can download them from https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/7782688/artifacts/download.

I am a bit confused by the error shown in the gitlab link above: 
`error: none of the selected packages contains these features: riscv`
Is it really coming from the modified benchmarking file? 🤔

**Comment by seadanda** [comment]
bot bench substrate-pallet --features=runtime-benchmarks --pallet=pallet_democracy

@ndkazu The default bench bot features includes one that doesn't exist in this pallet. Seems like a bad default, but this should work to override it.

**Review by muharem — APPROVED** [review]

**Comment by re-gius** [comment]
@ndkazu Are you sure the Polkadot address in your bio is correct?

**Pull request merged**